cockatrice/src/player.cpp

```diff
@@ -892,6 +902,18 @@ void Player::actViewGraveyard()
     static_cast<GameScene *>(scene())->toggleZoneView(this, "grave", -1);
 }
 
+void Player::actRevealRandomGraveyardCard()
+{
+    Command_RevealCards cmd;
+    QAction *action = static_cast<QAction *>(sender());
+    const int otherPlayerId = action->data().toInt();
+    if (otherPlayerId != -1)
+        cmd.set_player_id(otherPlayerId);
+    cmd.set_zone_name("grave");
```

Comment:
Here you can make this function generic by accepting the zone name as an argument